### PR TITLE
[SYCL] Windows MS-Compatibility changes

### PR DESCRIPTION
**Pull request by premanandrao**
Signed-off-by: Premanand M Rao <premanand.m.rao@intel.com>

### CODE REVIEW

**Comment by premanandrao** [comment]
This is a follow up for https://github.com/intel/llvm/pull/541

File: clang/test/SemaSYCL/delayed-parsing.cpp [hunk 1]
```diff
@@ -1,7 +1,9 @@
-// RUN: %clang_cc1 -std=c++1y -verify -fsyntax-only %s
-// RUN: %clang_cc1 -std=c++1y -verify -fsyntax-only %s -DDELAYED_TEMPLATE_PARSING -fdelayed-template-parsing
-// RUN: %clang %s -DDELAYED_TEMPLATE_PARSING -fsycl -c
-// REQUIRES: system-windows
+// RUN: %clang_cc1 -std=c++1y -verify -fsyntax-only %s -triple x86_64-unknown-windows-msvc
+// RUN: %clang_cc1 -std=c++1y -verify -fsyntax-only %s -DDELAYED_TEMPLATE_PARSING -fdelayed-template-parsing -triple x86_64-unknown-windows-msvc
+
```
Comment (bader) [inline clang/test/SemaSYCL/delayed-parsing.cpp]: Why this test is in `clang/test/SemaSYCL`? There no options enabling SYCL specific behavior.

I don't this this test is needed.
The only change #541 does is in the driver and it's covered by driver test and one pre-processor test `clang/test/Preprocessor/sycl-macro.cpp`. These two should be enough. 

I suggest removing the rest of the tests added in #541.

**Comment by premanandrao** [comment]
> Please, remove `clang/test/Preprocessor/sycl-macro2.cpp` and `clang/test/SemaSYCL/delayed-parsing2.cpp` as we agreed in #541.
> I also don't see any value in `clang/test/SemaSYCL/delayed-parsing.cpp` and suggest removing.

Done, thanks.

File: clang/test/Driver/windows-compat-sycl-args.cpp [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// RUN: %clang -std=c++1y -fsycl -c -v -- %s 2>&1 | FileCheck %s
```
[the remark this answers is not in the document]
Comment (premanandrao) [reply clang/test/Driver/windows-compat-sycl-args.cpp]: Also done.  @mdtoguchi , please review.

File: clang/test/Driver/windows-compat-sycl-args.cpp [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// RUN: %clang -### --sycl -c %s 2>&1 | FileCheck %s
+// RUN: %clang -### -fsycl-device-only -fsycl %s 2>&1 | FileCheck %s
```
Comment (mdtoguchi) [inline clang/test/Driver/windows-compat-sycl-args.cpp]: -fsycl-device-only is the same as --sycl (also -fsycl-device-only overrides -fsycl).  You can test against just -fsycl if you want to verify offload.

File: clang/test/Driver/windows-compat-sycl-args.cpp [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+// CHECK: -fdelayed-template-parsing
+// CHECK: -fms-compatibility
+// REQUIRES: system-windows
```
Comment (bader) [inline clang/test/Driver/windows-compat-sycl-args.cpp]: Please, set the triple with Windows OS component to make test OS agnostic.

File: clang/test/Driver/windows-compat-sycl-args.cpp [hunk 5]
```diff
@@ -0,0 +1,9 @@
+// RUN: %clang -### --sycl -c %s 2>&1 | FileCheck %s
```
Comment (bader) [inline clang/test/Driver/windows-compat-sycl-args.cpp]:
```suggestion
// RUN: %clang -### -fsycl-device-only -c %s 2>&1 | FileCheck %s
```

File: clang/test/Driver/windows-compat-sycl-args.cpp (hunk 3 above)
Comment (bader) [reply clang/test/Driver/windows-compat-sycl-args.cpp]: I think first command is enough to validate #541, so I think we can safely remove this line.

File: clang/test/Driver/windows-compat-sycl-args.cpp [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+// RUN: %clang -### -fsycl-device-only -fsycl %s 2>&1 | FileCheck %s
+// CHECK: -fdelayed-template-parsing
+// CHECK: -fms-compatibility
+// REQUIRES: system-windows
+// expected-no-diagnostics
+
```
Comment (romanovvlad) [inline clang/test/Driver/windows-compat-sycl-args.cpp]: Please put more details of what specific compatibility changes you doing and why in the commit message.

File: clang/test/Driver/windows-compat-sycl-args.cpp (hunk 4 above)
Comment (premanandrao) [reply clang/test/Driver/windows-compat-sycl-args.cpp]: @mdtoguchi , how do I set the triple with the driver so it is OS agnostic?

Comment (mdtoguchi) [reply clang/test/Driver/windows-compat-sycl-args.cpp]: You can add -target x86_64-unknown-windows-msvc (or something like that) to the command line.  The driver will behave according to the target specified.

**Comment by premanandrao** [comment]
@romanovvlad , thanks!

**Pull request merged**